app/models/manageiq/providers/openshift/container_manager/refresher.rb

```diff
@@ -3,9 +3,24 @@ module Openshift
     class ContainerManager::Refresher < ManageIQ::Providers::BaseManager::Refresher
       include ::EmsRefresh::Refreshers::EmsRefresherMixin
 
+      def self.entities
+        %w(routes projects)
+      end
+
       def parse_inventory(ems, _targets = nil)
-        all_entities = ems.with_provider_connection(&:all_entities)
-        all_entities.merge!(ems.with_provider_connection(:service => ManageIQ::Providers::Kubernetes::ContainerManager.ems_type, &:all_entities))
+        all_entities = ems.with_provider_connection { |client| fetch_entities(client, self.class.entities) }
+        fetch = proc do |client|
+          fetch_entities(
+            client,
+            ManageIQ::Providers::Kubernetes::ContainerManager::Refresher.entities
+          )
+        end
+        all_entities.merge!(
```

Comment:
@simon3z we talked about removing the merge call
I did not find a replacement for merge;
ems.with_provider_connection returns a hash. 
I need to return a new hash which is the union of two such hashes (they have disjoint keys, but it does not matter)
That's why I use merge, is there a better way @simon3z @blomquisg?